packages/passport/sdk/src/zkEvm/zkEvmProvider.ts

```diff
@@ -122,14 +130,12 @@ export class ZkEvmProvider implements Provider {
   }
 
   #handleLogout = () => {
-    const shouldEmitAccountsChanged = !!this.#zkEvmAddress;
-
     this.#ethSigner = undefined;
-    this.#zkEvmAddress = undefined;
+    this.#providerEventEmitter.emit(ProviderEvent.ACCOUNTS_CHANGED, []);
+  };
 
-    if (shouldEmitAccountsChanged) {
-      this.#providerEventEmitter.emit(ProviderEvent.ACCOUNTS_CHANGED, []);
-    }
+  #initialiseEthSignerOnLogin = (user: User) => {
+    if (isZkEvmUser(user)) this.#initialiseEthSigner(user);
```

Comment:
I think we should still initialise the eth signer if it's not a zkEvm user, otherwise the user will need to wait an additional 4-6~ seconds for the eth signer to be initialised during the `eth_requestAccounts` before their CFA can be generated.

Reply:
yeah good catch, by the looks though if they call `eth_requestAccounts` in this scenario, since they aren't a zkEvm registered user yet, the `this.#initialiseEthSigner(user)` will be called a second time regardless?

Reply:
addressed over slack, we agreed this is a perf issue not really a bug but one that had already existed so going to ship this. we're going to be revisiting the magic login in another initiative which will likely address this issue.